test/unit/test_client.py

```diff
@@ -1,24 +1,52 @@
+import warnings
+
 import pytest
 
 import ipfshttpclient
 
 
 def test_assert_version():
-	# Minimum required version
-	ipfshttpclient.assert_version("0.1.0", "0.1.0", "0.2.0", ["0.1.2"])
-	
+	with warnings.catch_warnings(record=True) as w:
+		# Cause all warnings to always be triggered.
+		warnings.simplefilter("always")
+
+		# Minimum required version
+		ipfshttpclient.assert_version("0.1.0", "0.1.0", "0.2.0", ["0.1.2"])
+
+		assert len(w) == 0
+
 	# Too high version
-	with pytest.raises(ipfshttpclient.exceptions.VersionMismatch):
+	with warnings.catch_warnings(record=True) as w:
+		# Cause all warnings to always be triggered.
+		warnings.simplefilter("always")
+
 		ipfshttpclient.assert_version("0.2.0", "0.1.0", "0.2.0", ["0.1.2"])
-	
+
+		assert len(w) == 1
+		assert issubclass(w[-1].category, ipfshttpclient.exceptions.VersionMismatch)
+
+
 	# Too low version
-	with pytest.raises(ipfshttpclient.exceptions.VersionMismatch):
+	with warnings.catch_warnings(record=True) as w:
+		# Cause all warnings to always be triggered.
+		warnings.simplefilter("always")
+
 		ipfshttpclient.assert_version("0.0.5", "0.1.0", "0.2.0", ["0.1.2"])
-	
+
+		assert len(w) == 1
+		assert issubclass(w[-1].category, ipfshttpclient.exceptions.VersionMismatch)
+
+
 	# Blacklisted version
-	with pytest.raises(ipfshttpclient.exceptions.VersionMismatch):
+	with warnings.catch_warnings(record=True) as w:
+		# Cause all warnings to always be triggered.
+		warnings.simplefilter("always")
+
 		ipfshttpclient.assert_version("0.1.2-1", "0.1.0", "0.2.0", ["0.1.2"])
 
+		assert len(w) == 1
+		assert issubclass(w[-1].category, ipfshttpclient.exceptions.VersionMismatch)
```

Comment:
Please use [`pytest.warns`](https://docs.pytest.org/en/stable/reference.html#pytest-warns) as it makes for more easily readable code.